dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/CompletionStageRxInvokerImpl.java

```diff
@@ -0,0 +1,210 @@
+    @Override
+    public CompletionStage<Response> put(Entity<?> entity) {
+        return put(entity, Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> put(Entity<?> entity, Class<T> responseType) {
+        return method(HttpMethod.PUT, entity, responseType);
+    }
+
+    @Override
+    public <T> CompletionStage<T> put(Entity<?> entity, GenericType<T> responseType) {
+        return method(HttpMethod.PUT, entity, responseType);
+    }
+
+    @Override
+    public CompletionStage<Response> post(Entity<?> entity) {
+        return post(entity, Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> post(Entity<?> entity, Class<T> responseType) {
+        return method(HttpMethod.POST, entity, responseType);
+    }
+
+    @Override
+    public <T> CompletionStage<T> post(Entity<?> entity, GenericType<T> responseType) {
+        return method(HttpMethod.POST, entity, responseType);
+    }
+
+    @Override
+    public CompletionStage<Response> delete() {
+        return delete(Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> delete(Class<T> responseType) {
+        return method(HttpMethod.DELETE, responseType);
+    }
+
+    @Override
+    public <T> CompletionStage<T> delete(GenericType<T> responseType) {
+        return method(HttpMethod.DELETE, responseType);
+    }
+
+    @Override
+    public CompletionStage<Response> head() {
+        return method(HttpMethod.HEAD);
+    }
+
+    @Override
+    public CompletionStage<Response> options() {
+        return options(Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> options(Class<T> responseType) {
+        return method(HttpMethod.OPTIONS, responseType);
+    }
+
+    @Override
+    public <T> CompletionStage<T> options(GenericType<T> responseType) {
+        return method(HttpMethod.OPTIONS, responseType);
+    }
+
+    @Override
+    public CompletionStage<Response> trace() {
+        return trace(Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> trace(Class<T> responseType) {
+        return method("TRACE", responseType);
+    }
+
+    @Override
+    public <T> CompletionStage<T> trace(GenericType<T> responseType) {
+        return method("TRACE", responseType);
+    }
+
+    @Override
+    public CompletionStage<Response> method(String name) {
+        return method(name, Response.class);
+    }
+
+    @Override
+    public CompletionStage<Response> method(String name, Entity<?> entity) {
+        return method(name, entity, Response.class);
+    }
+
+    @Override
+    public <T> CompletionStage<T> method(String name, Entity<?> entity, Class<T> responseType) {
+      //Liberty code change start
+        if (ex == null) {
+         // Run on Liberty executor thread and do not propagate the thread context
+            return (ManagedExecutor.builder().propagated(ThreadContext.NONE).build()).supplyAsync(() -> wc.sync().method(name, entity, responseType));            
```

Comment:
This will leak ManagedExecutor instances that will accumulate without ever being shut down, eventually causing out-of-memory.

It's also important to note that by using a ManagedExecutor with context propagation disabled, you still get context clearing.  I'm not sure if that is what you wanted here.  If so, then a better way to do this would be to construct a single, shared ManagedExecutor instance with context propagation disabled and reuse it.
Alternatively, if the interference with thread context is unwanted on this path, then the following would be the most efficient way to get what you are looking for:
com.ibm.ws.concurrent.mp.ManagedCompletableFuture.supplyAsync(Supplier<U>, Executor)
I'm not sure if the above method will be accessible to your code, but I'm certainly open to exporting an a helper interface or something that will give you access to it.

Reply:
Would this disable context propagation?
defaultManagedExecutor = ManagedExecutor.builder().propagated().build();

Reply:
Yes, `.propagated()` and `.propagated(ThreadContext.NONE)` are identical.  The `ThreadContext.NONE` constant is just there for users who prefer a more explicit way of writing it.  With all context propagation disabled on this executor, it will resort to clearing all thread context types for the duration of the completion stage action.